modules/tinymce/src/core/test/ts/browser/api/dom/RangeExpandTest.ts

```diff
@@ -0,0 +1,91 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Unicode } from '@ephox/katamari';
+import { TinyHooks, TinySelections } from '@ephox/mcagar';
+import { assert } from 'chai';
+
+import RangeUtils from 'tinymce/core/api/dom/RangeUtils';
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.core.api.dom.RangeExpandTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    indent: false,
+    base_url: '/project/tinymce/js/tinymce'
+  }, []);
+
+  const textEntry = 'A B ' + Unicode.nbsp + 'C D';
+
+  const makeBaseSelection = (editor: Editor, startOffset: number, endOffset: number, nodeOffset: number = 0) => {
```

Comment:
Should probably be called `makeBaseRange` also it could be path based so that you could feed it more exotic cases.